File: Spring2/BE/lego-manager-be/src/main/java/com/example/legomanagerbe/model/Lego.java
package com.example.legomanagerbe.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Lego {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    private Double price;

    private String releaseYear;

    private String numberOfPieces;

    private Integer quantity;

    @Column(columnDefinition = "text")
    private String imageUrl;

    @Column(name = "status_delete", columnDefinition = "BIT")
    private Integer statusDelete = 0;

    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;

}

File: Lego-Store/FE/src/app/model/lego.ts
import {Category} from './category';

export interface Lego {

  id?: number;

  name?: string;

  price?: number;

  releaseYear?: string;

  numberOfPieces?: string;

  quantity?: number;

  imageUrl?: string;

  imageBox?: string;

  category?: Category ;

}
